### ISSUE TITLE
simulation piq issues/questions

### ISSUE THREAD

**Issue by marquestrading** [betcode-org/flumine#766]
(Great framework, thanks for making it open source)

Two issues regarding simulation:
1. If there are two unmatched bets at the same price, the piq of the second bet is not updated until the first bet is matched so the piq of the second bet is incorrect. Suppose we are processing the simulation for bet 1:
In simulatedorder.py
	_calculate_process_traded() updates the _piq and returns traded_size even when the bet is not matched. 
	_process_traded() treats this as matched amount and deducts this from traded[traded_price]	
	When we then process bet 2, traded[traded_price] is 0, so the piq is not updated (but piq should change as amount has traded at this price)
	
Can _calculate_process_traded() return 0.0 instead for unmatched bets? Sorry have only recently started looking at flumine so Im not entirely sure about the downstream impact this change might have.
	
2. When config.simulation_available_prices = True, suppose a bet a comes in and matches your bet, simulation sets the matched price incorrectly. Suppose:
there are no bets at 1.01 and 1.02
you have a simulation back bet at 1.01 for 100GBP
somebody lays at 1.02 for 200 GBP, simulation sets the matched price to 1.02 instead of 1.01

I think it might be because in _process_available() we are passing in update["price"], instead of the order price?
                if update["price"] >= price:
                    self._calculate_process_available(
                        publish_time, update["price"], update["size"]
                    )
                                        
Apologies if I have missed something. I see there is issue#538 for smart matching but I dont see these issues in there. Please let me know if you need more information

**Comment by liampauling** [comment]
You are correct regarding Issue 2 and is an easy fix, Issue 1 isn't that simple due to the way the logic is currently coded.

I don't want to complicate things here mainly due to the fact it is in the simulation hot path, maybe after a coffee I will come up with something simple.

FYI join the [slack](https://join.slack.com/t/betcode-org/shared_invite/zt-25yz6dt1y-LHya5VzHLOzN3RZEQrSnrA) as I am sometimes slow to respond to GitHub issues.

**Comment by marquestrading** [comment]
Great thank you. Yep, will post on the slack going forward
